src/namespace.rs

```diff
@@ -4,19 +4,64 @@ use {XmlReader, Event, Element};
 use error::ResultPos;
 use std::io::BufRead;
 
+/// A namespace declaration. Can either bind a namespace to a prefix or define the current default
+/// namespace.
 #[derive(Clone)]
 struct Namespace {
-    prefix: Vec<u8>,
-    value: Vec<u8>,
-    element_name: Vec<u8>,
+    /// * `Some(prefix)` binds this namespace to `prefix`.
+    /// * `None` defines the current default namespace.
+    prefix: Option<Vec<u8>>,
+    /// The namespace name (the URI) of this namespace declaration.
+    ///
+    /// The XML standard specifies that an empty namespace value 'removes' a namespace declaration
+    /// for the extent of its scope. For prefix declarations that's not very interesting, but it is
+    /// vital for default namespace declarations. With `xmlns=""` you can revert back to the default
+    /// behaviour of leaving unqualified element names unqualified.
+    value: Option<Vec<u8>>,
+    /// Level of nesting at which this namespace was declared. The declaring element is included,
+    /// i.e., a declaration on the document root has `level = 1`.
+    /// This is used to pop the namespace when the element gets closed.
     level: i32,
 }
 
 impl Namespace {
+    /// Check whether this namespace declaration matches the **qualified element name** name.
+    /// Does not take default namespaces into account. The `matches_unqualified_elem` method is
+    /// responsible for unqualified element names.
+    ///
+    /// [W3C Namespaces in XML 1.1 (2006)](http://w3.org/TR/xml-names11/#scoping-defaulting)
     #[inline]
-    fn is_match(&self, name: &[u8]) -> bool {
-        let len = self.prefix.len();
-        name.len() > len && name[len] == b':' && &name[..len] == &*self.prefix
+    fn matches_qualified_elem(&self, name: &[u8]) -> bool {
+        if let Some(ref prefix) = self.prefix {
+            let len = prefix.len();
+            name.len() > len && name[len] == b':' && &name[..len] == &prefix[..]
+        } else {
+            false
+        }
+    }
+
+    /// A namespace declaration matches unqualified elements iff it is a default namespace
+    /// declaration (no prefix).
+    ///
+    /// [W3C Namespaces in XML 1.1 (2006)](http://w3.org/TR/xml-names11/#scoping-defaulting)
+    #[inline]
+    fn matches_unqualified_elem(&self) -> bool {
+        self.prefix.is_none()
+    }
+
+    /// Check whether this namespace declaration matches the (potentially prefixed) **attribute**
+    /// name. The default namespace does not apply to attributes. The "namespace name" of an
+    /// unprefixed attribute always has no value.
+    ///
+    /// [W3C Namespaces in XML 1.1 (2006)](http://w3.org/TR/xml-names11/#scoping-defaulting)
+    #[inline]
+    fn matches_qualified_attr(&self, name: &[u8]) -> bool {
```

Comment:
I may be wrong but I don't see the difference with the `match_qualified_elem` function.
I got that the unqualified is for elements only, but for qualified we could be DRY and have both function call the same `matches_qualified` function.